cilium/cmd/encrypt_status.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os/exec"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/cilium/cilium/api/v1/client/daemon"
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/command"
+	"github.com/cilium/cilium/pkg/common"
+	"github.com/prometheus/procfs"
+	"github.com/vishvananda/netlink"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// EncryptionModeDisabled captures enum value "Disabled"
+	EncryptionModeDisabled string = "Disabled"
+
+	// EncryptionModeIPsec captures enum value "IPsec"
+	EncryptionModeIPsec string = "IPsec"
+
+	// EncryptionModeWireguard captures enum value "Wireguard"
+	EncryptionModeWireguard string = "Wireguard"
+)
+
+type void struct{}
+
+var (
+	voidType    void
+	countErrors int
+)
+
+var encryptStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Display current state of encryption configurations",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium encrypt status")
+		getEncryptionMode()
+	},
+}
+
+func init() {
+	encryptCmd.AddCommand(encryptStatusCmd)
+	command.AddJSONOutput(encryptStatusCmd)
+}
+
+func getXfrmStats() (int, map[string]int) {
+	fs, err := procfs.NewDefaultFS()
+	if err != nil {
+		Fatalf("Cannot get a new proc FS: %s", err)
+	}
+	stats, err := fs.NewXfrmStat()
+	if err != nil {
+		Fatalf("Cannot get XFRM states from proc filesystem: %s", err)
+	}
+	v := reflect.ValueOf(stats)
+	errorMap := make(map[string]int)
+	for i := 0; i < v.NumField(); i++ {
+		name := v.Type().Field(i).Name
+		value := v.Field(i).Interface().(int)
+		if value != 0 {
+			countErrors += value
+			errorMap[name] = value
+		}
+	}
+	return countErrors, errorMap
+}
+
+func countUniqueIpsecKeys() int {
+	// trying to mimic set type data structure
+	// using void data type as struct{} because it does not use any memory
+	keys := make(map[string]void)
+	xfrmStates, _ := netlink.XfrmStateList(0)
+	for _, v := range xfrmStates {
+		keys[string(v.Aead.Key)] = voidType
+	}
+	return len(keys)
+}
+
+func maxSequenceNumber() string {
+	var maxSeqNum string
+	out, err := exec.Command("ip", "xfrm", "state").Output()
```

Comment:
To limit the check to Cilium rules, we could filter by the reqID. We always set it to 1: https://github.com/cilium/cilium/blob/v1.10.1/pkg/datapath/linux/ipsec/ipsec_linux.go#L505
```suggestion
	out, err := exec.Command("ip", "xfrm", "state", "show", "reqid", CiliumReqId).Output()
```